### PR TITLE
Keyvault Updates

### PR DESCRIPTION
**Pull request by g2vinay**
Adds Configuration + Logging
Lists Operation return Iterable  (Javadocs + Samples + Tests + README updated accordingly)
Tests Use Identity Library
Added preview version to poms.

### CODE REVIEW

File: keyvault/client/keys/README.md [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+
+## Getting started
+### Adding the package to your project
+
+Maven dependency for Azure Key Client library. Add it to your project's pom file.
+```xml
+<dependency>
+    <groupId>com.azure</groupId>
+    <artifactId>azure-keyvault-keys</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
```
Comment (JonathanGiles) [inline keyvault/client/keys/README.md]: This should be preview.1?

File: keyvault/client/keys/src/main/java/com/azure/keyvault/keys/KeyClient.java [hunk 2]
```diff
@@ -538,26 +469,27 @@ public VoidResponse purgeDeletedKey(String name) {
...
+     *   System.out.printf("Received key with name %s and type %s", keyWithMaterial.name(), keyWithMaterial.keyMaterial().kty());
+     * }
      * </pre>
      *
      * @return A {@link List} containing {@link KeyBase key} of all the keys in the vault.
      */
-    public List<KeyBase> listKeys() {
-        return service.getKeys(endpoint, DEFAULT_MAX_PAGE_RESULTS, API_VERSION, ACCEPT_LANGUAGE, CONTENT_TYPE_HEADER_VALUE).flatMapMany(r -> extractAndFetchKeys(r, Context.NONE)).collectList().block();
+    public Iterable<KeyBase> listKeys() {
+        return client.listKeys().collectList().block();
```
Comment (JonathanGiles) [inline keyvault/client/keys/src/main/java/com/azure/keyvault/keys/KeyClient.java]: Use Flux.toIterable: https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Flux.html#toIterable--

Comment (g2vinay) [reply keyvault/client/keys/src/main/java/com/azure/keyvault/keys/KeyClient.java]: updated in KeyClient and SecretClient list ops.

File: keyvault/client/keys/README.md (hunk 1 above)
Comment (g2vinay) [reply keyvault/client/keys/README.md]: yup, updated.

File: keyvault/client/secrets/src/samples/java/secrets/ListOperations.java [hunk 3]
```diff
@@ -28,16 +30,16 @@ public static void main(String[] args) throws IllegalArgumentException {
                 .credential(new AzureCredential())
                 .build();
 
-        // Let's create secrets holding storage and bank accounts credentials valid for 1 year. if the secret
+        // Let's create keys holding storage and bank accounts credentials valid for 1 year. if the secret
```
Comment (JonathanGiles) [inline keyvault/client/secrets/src/samples/java/secrets/ListOperations.java]: Your string replace has done too much, hasn't it? This is still a secrets sample?

**Pull request merged**